[tvOS][Bug] Fixes stale home content.

nokeeo opened the pull request:
When the tab bar controller appears on the screen the home view's content is refreshed.  This refreshes the content when it might be stale, like the user returning to the main page from the player.

Attaches an action on the tab bar coordinator's view.  Exposes the home coordinator's view model to refresh from the tab coordinator.

An alternative is to attach the onAppear to `HomeView` directly, however this action is fired on tab switch.  At the moment the content does not update on tab switch.

Fixes #904

LePips reviewed (changes requested):
Apologies, but this is not correct. We should not handle refreshing in the coordinators, only in the views. Please change to an implementation that does that.

nokeeo commented:
Updated.  After using this for a few days it was annoying after returning to the home view when browsing my library, for it to update and lose my spot.

Changed it so the home model subscribes to an invalidation notification.  I imagine there might be other cases where we want to invalidate, like returning from the background.

PTAL.

LePips commented on Shared/Objects/AppNotifications.swift:
```diff
@@ -0,0 +1,14 @@
+
+enum AppNotifications {
+    // Posted when the now playing data has likely become stale.
+    static let nowPlayingDidChange = Notification.Name(rawValue: "nowPlayingDidChange")
```
Don't create a new object to hold notification definitions, that's already in `SwiftfinNotifications.swift`.

LePips commented on Shared/ViewModels/HomeViewModel.swift:
```diff
@@ -26,15 +25,39 @@ final class HomeViewModel: ViewModel {
     var resumeItems: [BaseItemDto] = []
 
+    /// Indicates if the view the model is backing is visible. Setting this value may trigger a refresh.
+    var isVisible = false {
```
A view model should never hold a value whether a view is presented or not.

LePips commented on Swiftfin tvOS/Views/HomeView/HomeView.swift:
```diff
@@ -34,5 +34,11 @@ struct HomeView: View {
         }
         .edgesIgnoringSafeArea(.top)
         .edgesIgnoringSafeArea(.horizontal)
+        .onAppear(perform: {
```
Remove based on comment in the view model. Additionally, this is attached to a `Group` which would require more testing when changing states since I think the `onAppear/onDisappear` would be in an undeterministic with all of the views.

LePips reviewed (changes requested):
I was going to review this further, however this is still incorrect. I've always needed to fix how the state of the view model is handled and we should wait until then to implement this.

nokeeo commented:
> I've always needed to fix how the state of the view model is handled and we should wait until then to implement this.

@LePips I have some free time.   Happy to help with this if given direction.
